zipline/data/data_portal.py

```diff
@@ -1067,16 +1060,25 @@ def _get_daily_window_for_sid(self, asset, field, days_in_window,
             return_array = np.zeros((bar_count,))
 
         return_array[:] = np.NAN
-        sid = int(asset)
 
-        return_array[0:bar_count] = data[0].T[0]
-
-        self._apply_all_adjustments(
-            return_array,
-            sid,
-            days_in_window,
-            field,
-        )
+        start_date = self._get_asset_start_date(asset)
+        end_date = self._get_asset_end_date(asset)
+        day_slice = days_in_window.slice_indexer(start_date, end_date)
+        active_days = days_in_window[day_slice]
+
+        if active_days.shape[0]:
+            data = self._daily_bar_reader.history_window(field,
+                                                         active_days[0],
+                                                         active_days[-1],
+                                                         asset)
+            sid = int(asset)
```

Comment:
Ok, can remove. For what it is worth, at least in test history, the values passed down are integers. We should probably be using an Asset as a parameter to history in the tests, correct?